syntax = "proto3";

package sa01.memory.v1;

option go_package = "github.com/soma/sa01/memory/v1";

message MemoryRecord {
  string id = 1;
  string tenant = 2;
  string persona_id = 3;
  string content = 4;
  map<string, string> metadata = 5;
  int64 created_at_ms = 6;
}

message CreateMemoryRequest {
  string tenant = 1;
  string persona_id = 2;
  string content = 3;
  map<string, string> metadata = 4;
}

message CreateMemoryResponse {
  MemoryRecord record = 1;
}

message GetMemoryRequest {
  string id = 1;
}

message GetMemoryResponse {
  MemoryRecord record = 1;
}

message ListMemoriesRequest {
  string tenant = 1;
  string persona_id = 2;
  int32 limit = 3;
}

message ListMemoriesResponse {
  repeated MemoryRecord records = 1;
}

service MemoryService {
  rpc CreateMemory(CreateMemoryRequest) returns (CreateMemoryResponse);
  rpc GetMemory(GetMemoryRequest) returns (GetMemoryResponse);
  rpc ListMemories(ListMemoriesRequest) returns (ListMemoriesResponse);
}
